packages/react/src/components/Tile/next/Tile.js

```diff
@@ -0,0 +1,694 @@
+  ),
+
+  /**
+   * Specify the function to run when the ClickableTile is interacted with via a keyboard
+   */
+  handleKeyDown: deprecate(
+    PropTypes.func,
+    'The handleKeyDown prop for ClickableTile has been deprecated in favor of onKeyDown. It will be removed in the next major release.'
+  ),
+
+  /**
+   * The href for the link.
+   */
+  href: PropTypes.string,
+
+  /**
+   * `true` to use the light version. For use on $ui-01 backgrounds only.
+   * Don't use this to make tile background color same as container background color.
+   */
+  light: PropTypes.bool,
+
+  /**
+   * Specify the function to run when the ClickableTile is clicked
+   */
+  onClick: PropTypes.func,
+
+  /**
+   * Specify the function to run when the ClickableTile is interacted with via a keyboard
+   */
+  onKeyDown: PropTypes.func,
+
+  /**
+   * The rel property for the link.
+   */
+  rel: PropTypes.string,
+};
+
+export const SelectableTile = React.forwardRef(function SelectableTile(
+  {
+    value = 'value',
+    title = 'title',
+    selected = false,
+    tabIndex = 0,
+    light = false,
+    onClick = () => {},
+    onChange = () => {},
+    onKeyDown = () => {},
+    children,
+    id,
+    name,
+    // TODO: Remove iconDescription prop in the next major release
+    // eslint-disable-next-line no-unused-vars
+    iconDescription,
+    className,
+    handleClick,
+    handleKeyDown,
+    disabled,
+    ...rest
+  },
+  ref
+) {
+  const prefix = usePrefix();
+
+  // TODO: replace with onClick when handleClick prop is deprecated
+  const clickHandler = handleClick || onClick;
+
+  // TODO: replace with onKeyDown when handleKeyDown prop is deprecated
+  const keyDownHandler = handleKeyDown || onKeyDown;
+
+  const [isSelected, setIsSelected] = useState(selected);
+  const classes = cx(
+    `${prefix}--tile`,
+    `${prefix}--tile--selectable`,
+    {
+      [`${prefix}--tile--is-selected`]: isSelected,
+      [`${prefix}--tile--light`]: light,
+      [`${prefix}--tile--disabled`]: disabled,
+    },
+    className
+  );
+  const inputClasses = cx(`${prefix}--tile-input`, {
+    [`${prefix}--tile-input--checked`]: isSelected,
+  });
+
+  // TODO: rename to handleClick when handleClick prop is deprecated
+  function handleOnClick(evt) {
+    evt.preventDefault();
+    evt.persist();
+    setIsSelected(!isSelected);
+    clickHandler(evt);
+    onChange(evt);
+  }
+
+  // TODO: rename to handleKeyDown when handleKeyDown prop is deprecated
+  function handleOnKeyDown(evt) {
+    evt.persist();
+    if (matches(evt, [keys.Enter, keys.Space])) {
+      evt.preventDefault();
+      setIsSelected(!isSelected);
+      onChange(evt);
+    }
+    keyDownHandler(evt);
+  }
+
+  function handleChange(event) {
+    setIsSelected(event.target.checked);
+    onChange(event);
+  }
+
+  useEffect(() => {
+    setIsSelected(selected);
+  }, [selected]);
```

Comment:
We should update this to use this pattern for `getDerivedStateFromprops`: https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops